spec/system/nested_imageable_spec.rb

```diff
@@ -0,0 +1,264 @@
+require "rails_helper"
+
+describe "Nested imageable" do
+  factories = [
+    :budget,
+    :budget_investment,
+    :future_poll_question_option,
+    :proposal
+  ]
+
+  let(:factory) { factories.sample }
+  let!(:imageable) { create(factory) }
+  let!(:user) { create(:user, :level_two) }
+  let(:path) do
+    if factory == :budget
+      new_admin_budgets_wizard_budget_path
+    elsif factory == :budget_investment
+      [
+        new_budget_investment_path(budget_id: imageable.budget_id),
+        new_management_budget_investment_path(budget_id: imageable.budget_id)
+      ].sample
+    elsif factory == :future_poll_question_option
+      new_admin_option_image_path(option_id: imageable.id)
+    elsif factory == :proposal
+      [new_proposal_path, edit_proposal_path(imageable)].sample
+    end
```

Comment:
I'm wondering whether it makes sense to use a `case` here :thinking:.

```suggestion
    case factory
    when :budget
      new_admin_budgets_wizard_budget_path
    when :budget_investment
      [
        new_budget_investment_path(budget_id: imageable.budget_id),
        new_management_budget_investment_path(budget_id: imageable.budget_id)
      ].sample
    when :future_poll_question_option
      new_admin_option_image_path(option_id: imageable.id)
    when :proposal
      [new_proposal_path, edit_proposal_path(imageable)].sample
    end
```

Which one do you prefer?

Reply:
Maybe we could add the `Style/CaseLikeIf` rubocop rule :thinking:.